### ISSUE TITLE
The usage of layerDepth argument in SpriteBatch.Draw is not properly documented

### ISSUE THREAD

**Issue by KirylPrykhodzka** [MonoGame/MonoGame#8074]
I could not find any mention on the website that when using overloads of SpriteBatch.Draw that have layerDepth argument, layerDepth has to be between 0 and 1.
It would be nice if documentation / method summary specified it.

#### What version of MonoGame does the bug occur on:
- MonoGame 3.8.1

#### What operating system are you using:
- Windows

#### What MonoGame platform are you using:

- DesktopGL

**Comment by stromkos** [comment]
That is not true.  Some XNA documentation may present it that way, but monogame accepts any layer depth values.

The values are used as the Z component of the generated quad, so in a mixed 3D and Spritebatch environment, it is better to use a RenderTarget for the 2D items to render an invariant HUD.

**Comment by KirylPrykhodzka** [comment]
Upon some further investigation, it looks like I indeed was not totally correct. Please see more information below.

I have created a default MonoGame project for testing and added the following code:

```
private Color[] colors = new Color[]
{
    Color.Red,
    Color.Green,
    Color.Blue,
    Color.Yellow,
    Color.Purple,
};

protected override void Draw(GameTime gameTime)
{
    GraphicsDevice.Clear(Color.CornflowerBlue);

    _spriteBatch.Begin();

    for (int i = 0; i < 5; i++)
    {
        var texture = new Texture2D(GraphicsDevice, 1, 1);
        texture.SetData(new Color[] { colors[i] });
        _spriteBatch.Draw(
            texture,
            new Rectangle(100 + i * 10, 100 + i * 10, 100, 100),
            null,
            Color.White,
            0f,
            Vector2.Zero,
            SpriteEffects.None,
            i / 10);
    }

    _spriteBatch.End();

    base.Draw(gameTime);
}
```
Notice how layerDepth (last argument in `_spriteBatch.Draw` is `i / 10` which ensures that it is between 0.0 and 0.4.
When I run this code, the following image is displayed:

![выява](https://github.com/MonoGame/MonoGame/assets/38721357/e38fe886-e6ef-4c9a-9ebd-94ef1ae48f9e)

However, if I change layerDepth from `i / 10` to `i`, the following result is drawn:
![выява](https://github.com/MonoGame/MonoGame/assets/38721357/a4c1f215-b97b-4c7a-bca7-9a8778c8841e)

So I think there is something going on with layerDepth, or I am doing something wrong. Please help me understand this

The full code can be found here: https://github.com/KirylPrykhodzka/MonoGame_LayersTest

**Comment by KirylPrykhodzka** [comment]
Thank you for your answer @lithiumtoast . If I understand it correctly, layerDepth has to be between 0 and 1 in order for Draw method to work correctly (predictably). That's why I opened this issue.

**Comment by lithiumtoast** [comment]
>  If I understand it correctly, layerDepth has to be between 0 and 1 in order for Draw method to work correctly

Without changing the projection matrix yes. It is however possible to use `SpriteBatch` using a custom shader (effect) and use your own projection matrix as you wish. So then documenting that the layer depth always has to be `[0, 1]` would be misleading. It has to be only when using the default `SpriteEffect` (shader).

**Comment by nkast** [comment]
If you want to be more technical and presice, layerDepth has to be in the range of [GraphicsDevice.Viewport.MinDepth, GraphicsDevice.Viewport.MaxDepth].

By normalizing your depth in the range [0,1] and then multipling by the current viewport depth range will
guarantee that your code will 'just-work'.

**Comment by stromkos** [comment]
> If you want to be more technical and precise, layerDepth has to be in the range of [GraphicsDevice.Viewport.MinDepth, GraphicsDevice.Viewport.MaxDepth].

@nkast, this is not currently true, but has been corrected in my pull request.

> It's unfortunate that the XNA API does not allow to pass the projection matrix like it allow for the view matrix (`TransformMatrix` property).

@lithiumtoast , All of the parameters of the view matrix are now (as of my pull request) are available through the Viewport parameters. Yes, you are still stuck with an Orthogonal Perspective matrix.

**Comment by nkast** [comment]
Ok, some investigation is needed to verify one way or anothwr. I am curious about how an XNA project would behave.

**Comment by nkast** [comment]
You guys are right. Viewport has nothing to do with it.

layerDepth has to be in the range of [nearPlane, farPlane] of the projection matrix.

**Comment by stromkos** [comment]
@nkast, 

This change opens the door to scrolling without adjustment. But on the other side the hud would now need adjustment offsets.
